### PR TITLE
Splitting big values and other improvements

### PR DESCRIPTION
**Pull request by MathieuDutSik**
This PR is a piece of the PR https://github.com/linera-io/linera-protocol/pull/769 that causes more problems than expected.

The improvements done in this PR are:
* The `KeyValueStoreClientBigValue` allows to deal with clients that do not support large values.
* The tests have been significantly improved.
* The construction of test contexts has been simplified.
* A trait `MAX_VALUE_SIZE` has been added to the `KeyValueStoreClient` so that we know what is the maximal size supported.
* Resolve one bug in the `simplify` code for batches.
* Resolve a bug in the dynamoDb code when the journal entry was kept after write.

### CODE REVIEW

File: linera-views/src/batch.rs [hunk 1]
```diff
@@ -72,6 +72,20 @@ pub struct SimpleUnorderedBatch {
     pub insertions: Vec<(Vec<u8>, Vec<u8>)>,
 }
 
+impl SimpleUnorderedBatch {
+    /// Printing the simple unordered batch
+    pub fn print(&self) {
```
Comment (ma2bd) [inline linera-views/src/batch.rs:77]: Please let's use the `Debug` trait (as usual in Rust) for printing values during debugging.

**Review by ma2bd — COMMENTED** [review]
I understand that some bug fixes are related to existing code? Can you fix these bugs first? (Ideally in a separate PR and independently tested.)

Rationale: 1) fixing stuff before making them more complicated 2) getting in the habit of making bug-fixes cherry-pickable.

File: linera-views/src/batch.rs (hunk 1 above)
Comment (MathieuDutSik) [reply linera-views/src/batch.rs:77]: I tended to like this more than the Debug but I will follow your advice and remove it.

**Comment by MathieuDutSik** [comment]
The PR is still too big and then split into smaller blocks.

**Pull request closed without merging**